mllib/src/main/scala/org/apache/spark/mllib/api/python/PythonMLLibAPI.scala

```diff
@@ -1105,6 +1108,59 @@ private[python] class PythonMLLibAPI extends Serializable {
     Statistics.kolmogorovSmirnovTest(data, distName, paramsSeq: _*)
   }
 
+  /**
+   * Wrapper around DistributedMatrices.rowMatrix factory method.
+   */
+  def createRowMatrix(rows: JavaRDD[Vector], numRows: Long, numCols: Int): RowMatrix = {
+    DistributedMatrices.rowMatrix(rows.rdd, numRows, numCols)
+  }
+
+  /**
+   * Wrapper around DistributedMatrices.indexedRowMatrix factory method.
+   */
+  def createIndexedRowMatrix(rows: DataFrame, numRows: Long, numCols: Int): IndexedRowMatrix = {
+    // We use DataFrames for serialization of IndexedRows from Python, so map each Row in the
+    // DataFrame back to an IndexedRow.
+    val indexedRows = rows.map { row =>
+      val index = row.getLong(0)
+      val vector = row.getAs[Vector](1)
+      IndexedRow(index, vector)
```

Comment:
I am used to writing this in a line like

    rows.map { case Row(ind: Long, vec: Vector) => IndexedRow(ind, vec) }

But that might be personal preference